### ISSUE TITLE
Trainer.predict using customized model.predict function?

### ISSUE THREAD

**Issue by iamlockelightning** [huggingface/transformers#11923]
I am using the [Trainer](https://huggingface.co/transformers/main_classes/trainer.html) to train a sentence-bert model with triplet-loss. Then I want to do some inference. How to call Trainer.predict using custom model.predict function?

I use `model.forward()` to calculate loss in training stage. But I want to use a customized `model.predict()` to calculate prediction results based on `model.forward()` (e.g., model.forward() -> embedding -> other method to calculate prediction instead of the loss function)

I saw the `prediction_step()` function just called `outputs = model(**inputs)` to get `(loss, logits, labels)`

Is there any good method to do that?

**Comment by shabie** [comment]
Cant you just subclass the Trainer class and write your own `predict`?
